addons/binding/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/handler/HyperionHandler.java

```diff
@@ -0,0 +1,324 @@
+            List<ActiveEffect> activeEffects = info.getActiveEffects();
+            List<Effect> effects = info.getEffects();
+            updateEffect(activeEffects, effects);
+
+            // update transform
+            List<Transform> transform = info.getTransform();
+            updateTransform(transform);
+        }
+    }
+
+    private void updateTransform(List<Transform> transform) {
+        Transform defaultTransform = transform.stream() // convert list to stream
+                .findFirst().orElse(null);
+
+        if (defaultTransform == null) {
+            updateState(CHANNEL_BRIGHTNESS, UnDefType.NULL);
+        } else {
+            double luminanceGain = defaultTransform.getLuminanceGain();
+            if (luminanceGain >= 0.0 && luminanceGain <= 1.0) {
+                PercentType luminanceGainPercentType = new PercentType((int) (luminanceGain * 100));
+                updateState(CHANNEL_BRIGHTNESS, luminanceGainPercentType);
+            }
+        }
+    }
+
+    private void updateEffect(List<ActiveEffect> activeEffects, List<Effect> effects) {
+        ActiveEffect effect = activeEffects.stream() // convert list to stream
+                .findFirst().orElse(null);
+        if (effect == null) {
+            updateState(CHANNEL_EFFECT, UnDefType.NULL);
+        } else {
+            String path = effect.getScript();
+            Effect effectDescription = effects.stream() // convert list to stream
+                    .filter(effectCandidate -> effectCandidate.getScript().equals(path)).findFirst().orElse(null);
+            String effectName = effectDescription.getName();
+            updateState(CHANNEL_EFFECT, new StringType(effectName));
+        }
+    }
+
+    private void updateColor(List<ActiveLedColor> activeColors) {
+        ActiveLedColor color = activeColors.stream() // convert list to stream
+                .findFirst().orElse(null);
+        if (color == null) {
+            updateState(CHANNEL_COLOR, UnDefType.NULL);
+        } else {
+            List<Integer> rgbVals = color.getRGBValue();
+            int r = rgbVals.get(0);
+            int g = rgbVals.get(1);
+            int b = rgbVals.get(2);
+            HSBType hsbType = HSBType.fromRGB(r, g, b);
+            updateState(CHANNEL_COLOR, hsbType);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Hyperion thing handler.");
+        try {
+
+            Configuration config = thing.getConfiguration();
+            address = (String) config.get(PROP_HOST);
+            port = ((BigDecimal) config.get(PROP_PORT)).intValue();
+            refreshInterval = ((BigDecimal) config.get(PROP_POLL_FREQUENCY)).intValue();
+            priority = ((BigDecimal) config.get(PROP_PRIORITY)).intValue();
+
+            connection = new JsonTcpConnection(address, port);
+            connectFuture = scheduler.scheduleWithFixedDelay(connectionJob, 0, refreshInterval, TimeUnit.SECONDS);
```

Comment:
This may cause the behaviour described in issue #3117 

According to [JavaDoc](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html):
> Beware however that expiration of a relative delay need not coincide with the current Date at which the task is enabled due to network time synchronization protocols, clock drift, or other factors. The Executors class provides convenient factory methods for the ScheduledExecutorService implementations provided in this package.

So I guess this must not be 0.

Reply:
Is there a way to define the minimum value for a thing-type property? I'd prefer to validate it there than in initialize()

Reply:
Yes, there is. Have a look at https://www.eclipse.org/smarthome/documentation/development/bindings/xml-reference.html#xml-structure-for-configuration-descriptions
There are `min`, `max` & `step` attributes.